src/process/posix/SDL_posixprocess.c

```diff
@@ -0,0 +1,298 @@
+            goto posix_spawn_fail_all;
+        }
+    } else if (redirect_stderr) {
+        if (posix_spawn_file_actions_adddup2(&fa, STDOUT_FILENO, STDERR_FILENO) != 0) {
+            SDL_SetError("posix_spawn_file_actions_adddup2 failed: %s", strerror(errno));
+            goto posix_spawn_fail_all;
+        }
+    }
+
+    // Spawn the new process
+    if (posix_spawnp(&data->pid, args[0], &fa, &attr, args, env) != 0) {
+        SDL_SetError("posix_spawn failed: %s", strerror(errno));
+        goto posix_spawn_fail_all;
+    }
+    SDL_SetNumberProperty(process->props, SDL_PROP_PROCESS_PID_NUMBER, data->pid);
+
+    if (enable_stdin) {
+        if (!SetupStream(process, stdin_pipe[WRITE_END], "wb", SDL_PROP_PROCESS_STDIN_POINTER)) {
+            close(stdin_pipe[WRITE_END]);
+        }
+        close(stdin_pipe[READ_END]);
+    }
+
+    if (enable_stdout) {
+        if (!SetupStream(process, stdout_pipe[READ_END], "rb", SDL_PROP_PROCESS_STDOUT_POINTER)) {
+            close(stdout_pipe[READ_END]);
+        }
+        close(stdout_pipe[WRITE_END]);
+    }
+
+    if (enable_stderr) {
+        if (!SetupStream(process, stderr_pipe[READ_END], "rb", SDL_PROP_PROCESS_STDERR_POINTER)) {
+            close(stderr_pipe[READ_END]);
+        }
+        close(stderr_pipe[WRITE_END]);
+    }
+
+    posix_spawn_file_actions_destroy(&fa);
+    posix_spawnattr_destroy(&attr);
+
+    return true;
+
+    /* --------------------------------------------------------------------- */
+
+posix_spawn_fail_all:
+    posix_spawn_file_actions_destroy(&fa);
+
+posix_spawn_fail_attr:
+    posix_spawnattr_destroy(&attr);
+
+posix_spawn_fail_none:
+    if (stdin_pipe[READ_END] >= 0) {
+        close(stdin_pipe[READ_END]);
+    }
+    if (stdin_pipe[WRITE_END] >= 0) {
+        close(stdin_pipe[WRITE_END]);
+    }
+    if (stdout_pipe[READ_END] >= 0) {
+        close(stdout_pipe[READ_END]);
+    }
+    if (stdout_pipe[WRITE_END] >= 0) {
+        close(stdout_pipe[WRITE_END]);
+    }
+    if (stderr_pipe[READ_END] >= 0) {
+        close(stderr_pipe[READ_END]);
+    }
+    if (stderr_pipe[WRITE_END] >= 0) {
+        close(stderr_pipe[WRITE_END]);
+    }
+    return false;
+}
+
+bool SDL_SYS_KillProcess(SDL_Process *process, SDL_bool force)
+{
+    int ret = kill(process->internal->pid, force ? SIGKILL : SIGTERM);
+    if (ret == 0) {
+        return true;
+    } else {
+        return SDL_SetError("Could not kill(): %s", strerror(errno));
+    }
+}
+
+bool SDL_SYS_WaitProcess(SDL_Process *process, SDL_bool block, int *exitcode)
+{
+    int wstatus = 0;
+    int ret = waitpid(process->internal->pid, &wstatus, block ? 0 : WNOHANG);
+
+    if (ret < 0) {
+        return SDL_SetError("Could not waitpid(): %s", strerror(errno));
+    }
+
+    if (ret == 0) {
+        SDL_ClearError();
+        return false;
+    }
+
+    if (WIFEXITED(wstatus)) {
+        *exitcode = WEXITSTATUS(wstatus);
+    }
+
+    if (WIFSIGNALED(wstatus)) {
```

Comment:
```suggestion
    }
    else if (WIFSIGNALED(wstatus)) {
```

and then we can have a final `else` clause for the case where some secret third thing has happened.